src/components/TimeseriesChartExport/TimeseriesChartExport.tsx

```diff
@@ -0,0 +1,409 @@
+import { Aggregate, DatapointsMultiQuery, InternalId } from '@cognite/sdk';
+import { TimeSeries } from '@cognite/sdk/dist/src/resources/classes/timeSeries';
+import { TimeSeriesList } from '@cognite/sdk/dist/src/resources/classes/timeSeriesList';
+import {
+  Alert,
+  Button,
+  Checkbox,
+  DatePicker,
+  Form,
+  Input,
+  Modal,
+  Radio,
+} from 'antd';
+import { RangePickerValue } from 'antd/lib/date-picker/interface';
+import { FormComponentProps } from 'antd/lib/form';
+import { ColProps } from 'antd/lib/grid';
+import moment, { Moment } from 'moment-timezone';
+import React, { SyntheticEvent, useContext, useEffect, useState } from 'react';
+import { ClientSDKContext } from '../../context/clientSDKContext';
+import { withDefaultTheme } from '../../hoc/withDefaultTheme';
+import { AnyIfEmpty, PureObject } from '../../interfaces';
+import { defaultTheme } from '../../theme/defaultTheme';
+import { datapointsToCSV, Delimiters, downloadCSV } from '../../utils/csv';
+import { getGranularityInSeconds } from '../../utils/utils';
+import { ComplexString } from '../common/ComplexString/ComplexString';
+
+export type FetchCSVCall = (
+  request: DatapointsMultiQuery,
+  opts: CsvParseOptions
+) => Promise<string>;
+
+export type FetchTimeseriesCall = (
+  ids: InternalId[]
+) => Promise<TimeSeriesList>;
+
+export interface CsvParseOptions {
+  aggregate: Aggregate;
+  delimiter: Delimiters;
+  readableDate: boolean;
+  granularity: string;
+}
+
+export interface FormItemLayout {
+  labelCol: ColProps;
+  wrapperCol: ColProps;
+}
+
+export interface ChartExportFormFields {
+  range: Moment[];
+  granularity: string;
+  delimiter: Delimiters;
+  readableDate: boolean;
+}
+
+export interface TimeseriesChartExportProps extends FormComponentProps {
+  timeseries: InternalId[];
+  granularity: string;
+  defaultRange: number[];
+  visible: boolean;
+  modalWidth?: number;
+  cellLimit?: number;
+  downloadAsSvg?: () => void;
+  fetchCSV?: FetchCSVCall;
+  hideModal?: () => void;
+  retrieveTimeseries?: FetchTimeseriesCall;
+  onError?: (err: any) => void;
+  onSuccess?: () => void;
+  formItemLayout?: FormItemLayout;
+  theme?: AnyIfEmpty<{}>;
+  strings?: PureObject;
+}
+
+const defaultStrings: PureObject = {
+  title: 'Export chart data',
+  labelRange: 'Range',
+  labelGranularity: 'Label Granularity',
+  labelGranularityHelp: 'Example inputs: 15s, 1m, 5h, 2d',
+  formatTimestamp: 'Format timestamp?',
+  formatTimestampHelp: 'e.g. 2018-04-02 12:20:20',
+  delimiterLabel: 'Select delimiter',
+  delimiterHelp: 'The character that will separate your data fields',
+  csvDownload: 'Download as CSV',
+  csvDownloadInProgress: 'Download as CSV',
+  closeBtn: 'Close',
+  imageDownloadLabel: 'Image download',
+  imageDownloadBtn: 'Download as SVG',
+  cellLimitErr:
+    'You hit the limit of {{ cellLimit }} datapoints - some data may be omitted',
+};
+
+// TODO: Check tree shacking for this component
```

Comment:
Just thought I'd double check whether this Todo is intended to be brought into the code base. I'm not exactly sure what component is meant to be checked for tree shaking?

Reply:
that is more if an improvement to consider later, related not only to this specific case as I understood

Reply:
We need to review antd imports, how they affect bundle size of gearbox. Just leave it here as a placeholder